airbyte-integrations/connectors/source-mysql/src/main/kotlin/MySqlSourceExceptionHandler.kt

```diff
@@ -31,8 +31,73 @@ class MySqlSourceExceptionHandler : ConnectorExceptionHandler() {
                 failureType = FailureType.TRANSIENT,
                 externalMessage = "Can not read data from MySQL server",
                 sampleInternalMessage =
-                    "java.io.EOFException: Can not read response from server. Expected to read X bytes, read Y bytes before connection was unexpectedly lost."
-            )
+                    "java.io.EOFException: Can not read response from server. Expected to read X bytes, read Y bytes before connection was unexpectedly lost.",
+            ),
+        )
+
+        add(
+            ConnectorErrorProfile(
+                errorClass = "MySQL Hikari Connection Error",
+                regexMatchingPattern = ".*connection is not available, request timed out after*",
+                failureType = FailureType.TRANSIENT,
+                externalMessage = "Database read failed due to connection timeout, will retry.",
+                sampleInternalMessage =
+                    "java.sql.SQLTransientConnectionException: HikariPool-x - Connection is not available, request timed out after xms",
+                referenceLinks = listOf("https://github.com/airbytehq/airbyte/issues/41614"),
+            ),
+        )
+
+        add(
+            ConnectorErrorProfile(
+                errorClass = "MySQL Timezone Error",
+                regexMatchingPattern = ".*is unrecognized or represents more than one time zone*",
+                failureType = FailureType.CONFIG,
+                externalMessage =
+                    "Please configure your database with the correct timezone found in the detailed error message. " +
+                        "Please refer to the following documentation: https://dev.mysql.com/doc/refman/8.4/en/time-zone-support.html",
+                sampleInternalMessage =
```

Comment:
@theyueli  What is this field supposed to be used for? I can't find any usages in the code base?

Reply:
correct. We used this field to store a sample message which provides a bit more context for this error. This should be an optional field to fill. The purpose is similar to the reference link field which provides links to github/sentry issues.